applications/ColossalQA/colossalqa/retrieval_conversation_universal.py

```diff
@@ -0,0 +1,138 @@
+"""
+Multilingual retrieval based conversation system
+"""
+from typing import List
+
+from colossalqa.data_loader.document_loader import DocumentLoader
+from colossalqa.mylogging import get_logger
+from colossalqa.retrieval_conversation_en import EnglishRetrievalConversation
+from colossalqa.retrieval_conversation_zh import ChineseRetrievalConversation
+from colossalqa.retriever import CustomRetriever
+from colossalqa.text_splitter import ChineseTextSplitter
+from colossalqa.utils import detect_lang_naive
+from langchain.embeddings import HuggingFaceEmbeddings
+from langchain.text_splitter import RecursiveCharacterTextSplitter, TextSplitter
+
+logger = get_logger()
+
+
+class UniversalRetrievalConversation:
+    """
+    Wrapper class for bilingual retrieval conversation system
+    """
+
+    def __init__(
+        self,
+        embedding_model_path: str = "moka-ai/m3e-base",
+        embedding_model_device: str = "cpu",
+        zh_model_path: str = None,
+        zh_model_name: str = None,
+        en_model_path: str = None,
+        en_model_name: str = None,
+        sql_file_path: str = None,
+        files_zh: List[List[str]] = None,
+        files_en: List[List[str]] = None,
+        text_splitter_chunk_size=100,
+        text_splitter_chunk_overlap=10,
+    ) -> None:
+        """
+        Warpper for multilingual retrieval qa class (Chinese + English)
+        Args:
+            embedding_model_path: local or huggingface embedding model
+            embedding_model_device:
+            files_zh: [[file_path, name_of_file, separator],...] defines the files used as supporting documents for Chinese retrieval QA
+            files_en: [[file_path, name_of_file, separator],...] defines the files used as supporting documents for English retrieval QA
+        """
+        self.embedding = HuggingFaceEmbeddings(
+            model_name=embedding_model_path,
+            model_kwargs={"device": embedding_model_device},
+            encode_kwargs={"normalize_embeddings": False},
+        )
+        print("Select files for constructing Chinese retriever")
+        docs_zh = self.load_supporting_docs(
+            files=files_zh,
+            text_splitter=ChineseTextSplitter(
+                chunk_size=text_splitter_chunk_size, chunk_overlap=text_splitter_chunk_overlap
+            ),
+        )
+        # Create retriever
+        self.information_retriever_zh = CustomRetriever(
+            k=3, sql_file_path=sql_file_path.replace(".db", "_zh.db"), verbose=True
+        )
+        self.information_retriever_zh.add_documents(
+            docs=docs_zh, cleanup="incremental", mode="by_source", embedding=self.embedding
+        )
+
+        print("Select files for constructing English retriever")
```

Comment:
remove debugging info.

Reply:
this line is a part of the interactive program. It prompts the user to select files for constructing RAG.